### PR TITLE
Add `skip_wait_on_job_termination` option for dataflow job resources

### PR DESCRIPTION
**Pull request by n-oden**
This addresses https://github.com/hashicorp/terraform-provider-google/issues/10559

See also https://github.com/hashicorp/terraform-provider-google-beta/pull/3868

### CODE REVIEW

**Comment by n-oden** [comment]
@ScottSuarez Happy new year!  Did you have any thoughts on this one?

**Comment by ScottSuarez** [comment]
I think I understand the use case but could you elaborate or go into more detail on your change?

**Comment by n-oden** [comment]
@ScottSuarez https://github.com/hashicorp/terraform-provider-google/issues/10559 goes into a lot more detail; the 5s summary is "sometimes draining streaming dataflow jobs takes a ridiculously long time" -- minutes to hours depending on what sort of internal windowing/watermark strategy your job is using.  This means that potentially your terraform workspace is locked for that entire time while waiting for the job to drain.

With this in place, operators will have the option of waiting only until the job reaches the DRAINING state before moving on.  Obviously this has the caveat that if you try to create a replacement job with the same name before the job actually exits it will fail, hence the digression about random_id in the doc examples.

**Comment by n-oden** [comment]
@rileykarson happy new year!  Thoughts on this?

**Comment by n-oden** [comment]
@ScottSuarez @rileykarson nudge?  If this isn't an approach to #10559 that you like, please let me know.

**Comment by ScottSuarez** [comment]
taking over review here. I'll give this a look over soon

File: google/resource_dataflow_job.go [hunk 1]
```diff
@@ -471,6 +478,11 @@ func resourceDataflowJobDelete(d *schema.ResourceData, meta interface{}) error {
 
 	// Wait for state to reach terminal state (canceled/drained/done)
 	_, ok := dataflowTerminalStatesMap[d.Get("state").(string)]
+	if !ok {
+		if d.Get("skip_wait_on_job_termination").(bool) {
+			_, ok = dataflowTerminatingStatesMap[d.Get("state").(string)]
+		}
+	}
```
Comment (ScottSuarez) [inline google/resource_dataflow_job.go:485]: This logic is repeated several times. Can you extrapolate this to a function

```golang
func isDataflowJobResourceDeleted(state string, skipWait bool) bool {
	_, stopQuery := dataflowTerminalStatesMap[state]
	if !stopQuery && skipWait {
		_, stopQuery = dataflowTerminatingStatesMap[d.Get("state").(string)]
	}
	return stopQuery
}
```

**Review by ScottSuarez — CHANGES_REQUESTED** [review]
Could you also add a test case to the testsuite for this resource and ensure it passes

https://github.com/hashicorp/terraform-provider-google/blob/main/google/resource_dataflow_job_test.go

**Comment by n-oden** [comment]
All reasonable requests!  I'll try to get it sorted out both here and in the beta-provider PR on monday or tuesday.

**Comment by ScottSuarez** [comment]
Can you make the changes against the upstream repository https://github.com/GoogleCloudPlatform/magic-modules

We use this to generate the two version of the provider. Tag me on the PR when you open it

**Comment by n-oden** [comment]
continued in https://github.com/GoogleCloudPlatform/magic-modules/pull/5844

**Pull request closed without merging**